connectors/src/connectors/google_drive/index.ts

```diff
@@ -304,43 +308,212 @@ export async function retrieveGoogleDriveConnectorPermissions(
     logger.error({ connectorId }, "Connector not found");
     return new Err(new Error("Connector not found"));
   }
+  const authCredentials = await getAuthObject(c.connectionId);
+  if (filterPermission === "read") {
+    if (parentInternalId === null) {
+      // Return the list of folders explicitly selected by the user.
+      const folders = await GoogleDriveFolders.findAll({
+        where: {
+          connectorId: connectorId,
+        },
+      });
 
-  const folders = await GoogleDriveFolders.findAll({
-    where: {
-      connectorId: connectorId,
-    },
-  });
+      const driveClient = await getDriveClient(authCredentials);
+
+      const resources: ConnectorResource[] = await Promise.all(
+        folders.map(async (f): Promise<ConnectorResource> => {
+          const folder = await driveClient.files.get({
```

Comment:
As per IRL, let's not got to GDrive here and show what we do have so look locally

Reply:
Revised IRL conversation: we stick to Gdrive for now.